src/lib/button/button.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, Input, Output, EventEmitter, ViewChild, ElementRef,
+        ViewEncapsulation, ChangeDetectionStrategy, OnInit, HostBinding, AfterViewInit, OnChanges } from '@angular/core';
+
+@Component({
+    selector: 'button[cui-button], a[cui-button], input[cui-button]',
+    exportAs: 'cuiButton',
+    template: `
+    <div #button>
+      <cui-loading *ngIf='loading'></cui-loading>
+      <span [ngStyle]="{'opacity': opacity}">
+        <ng-content></ng-content>
+      </span>
+    </div>
+    `,
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+  })
+  export class ButtonComponent implements OnInit, AfterViewInit, OnChanges {
+
+    @Input() public active: boolean = false;
+    @Input() public ariaLabel: string = '';
+    @Input() public ariaLabelledBy: string = '';
+    @Input() public circle: boolean = false;
+    @Input() public className: string = '';
+    @Input() public color: string = '';
+    @Input() public disabled: boolean = false;
+    @Input() public expand: boolean = false;
+    @Input() public href: string = '';
+    @Input() public loading: boolean = false;
+    @Input() public removeStyle: boolean = false;
+    @Input() public size: string = '36';
+    @Input() public type: string = 'button';
+
+    @HostBinding('attr.active') get activeOption() { return this.active || null; }
+    @HostBinding('attr.ariaLabel') get ariaLabelOption() { return this.ariaLabel || null; }
+    @HostBinding('attr.ariaLabelledBy') get ariaLabelledByOption() { return this.ariaLabelledBy || null; }
```

Comment:
I do not suggest having aria label and aria labelledby here. Even if you guys decide to have it,

it should have been `aria-labelledby` instead of `ariaLabelledBy` which is a React prop for it

```suggestion
    @HostBinding('attr.aria-labelledby') get ariaLabelledByOption() { return this.ariaLabelledBy || null; }
```

Reply:
Agreed, we need to be careful when we use the React components as a guide, to use the actual HTML attributes and not the JSX versions.